src/Adapters/ApproversCategoriesLookup.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace ProfessionalWiki\PageApprovals\Adapters;
+
+use ProfessionalWiki\PageApprovals\Application\ApproversCategoriesLookupInterface;
+use Wikimedia\Rdbms\IDatabase;
+
+class ApproversCategoriesLookup implements ApproversCategoriesLookupInterface {
+
+	private const APPROVERS_GROUP = 'approvers';
+
+	public function __construct(
+		private readonly IDatabase $database
+	) {
+	}
+
+	/**
+	 * @return array<array{userId: int, username: string, categories: string[]}>
+	 */
+	public function getApproversWithCategories(): array {
```

Comment:
I am not sure if we should move this to ApproversRepository `getApproversWithCategories` but since that interface is being used in my other places we will end up having it in places where we don't need.